File: backend/app/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.notification import NotificationType


class NotificationResponse(BaseModel):
    id: int
    issue_id: int | None
    issue_title: str | None
    title: str
    message: str
    notification_type: NotificationType
    is_read: bool
    created_at: datetime
    read_at: datetime | None

    model_config = ConfigDict(from_attributes=True)


class NotificationListResponse(BaseModel):
    items: list[NotificationResponse]
    total: int
    page: int
    page_size: int
    total_pages: int


class UnreadCountResponse(BaseModel):
    count: int

File: frontend/src/types/notification.ts
export type NotificationType =
  | 'issue_status_updated'
  | 'issue_resolved'
  | 'high_priority_issue'

export type Notification = {
  id: number
  issue_id: number | null
  issue_title: string | null
  title: string
  message: string
  notification_type: NotificationType
  is_read: boolean
  created_at: string
  read_at: string | null
}

export type NotificationListResponse = {
  items: Notification[]
  total: number
  page: number
  page_size: number
  total_pages: number
}
